pkg/registry/rbac/clusterrole/registry.go

```diff
@@ -62,6 +61,6 @@ type AuthorizerAdapter struct {
 }
 
 // GetClusterRole returns the corresponding ClusterRole by name
-func (a AuthorizerAdapter) GetClusterRole(name string) (*rbacv1.ClusterRole, error) {
-	return a.Registry.GetClusterRole(genericapirequest.NewContext(), name, &metav1.GetOptions{})
+func (a AuthorizerAdapter) GetClusterRole(ctx context.Context, name string) (*rbacv1.ClusterRole, error) {
+	return a.Registry.GetClusterRole(ctx, name, &metav1.GetOptions{})
```

Comment:
Added here and everywhere else too.

Reply:
> Added here and everywhere else too.

Adding everywhere means that call chains that go through multiple layers wrap context a bunch of times... the linked-list implementation of context value getters means that makes calls more expensive for all other GetValue() calls in the context.

Let's limit this to just the implementation where we call to the etcd registries rather than re-wrapping at every layer

a.Registry.GetClusterRole / a.Registry.ListClusterRoleBindings / etc